engine/access/rest/websockets/controller.go

```diff
@@ -202,20 +228,21 @@ func (c *Controller) handleListSubscriptions(ctx context.Context, msg models.Lis
 }
 
 func (c *Controller) shutdownConnection() {
-	defer close(c.communicationChannel)
-	defer func(conn *websocket.Conn) {
-		if err := c.conn.Close(); err != nil {
-			c.logger.Error().Err(err).Msg("error closing connection")
-		}
-	}(c.conn)
+	c.shutdownOnce.Do(func() {
```

Comment:
`shutdownConnection` is only called in one place and should never be called multiple times. I don't think we need to synchronization here.